include/AsyncTraceDispatcher.h

```diff
@@ -0,0 +1,131 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#ifndef __ASYNCTRACEDISPATCHER_H__
+#define __ASYNCTRACEDISPATCHER_H__
+
+#include <atomic>
+#include <condition_variable>
+#include <memory>
+#include <string>
+#include <thread>
+
+#include "DefaultMQProducer.h"
+
+#include "TraceHelper.h"
+
+#include <memory>
+#include "ClientRPCHook.h"
+#include "TraceDispatcher.h"
+namespace rocketmq {
+
+class AsyncTraceDispatcher : public TraceDispatcher, public enable_shared_from_this<AsyncTraceDispatcher> {
```

Comment:
`public enable_shared_from_this<AsyncTraceDispatcher>`  seems can be `private` here right?